packages/pricing/src/repositories/pricing.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  CalculatedPriceSetDTO,
+  Context,
+  PricingContext,
+  PricingFilters,
+} from "@medusajs/types"
+import { MedusaError, MikroOrmBase } from "@medusajs/utils"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+import { PricingRepositoryService } from "../types"
+
+export class PricingRepository
+  extends MikroOrmBase
+  implements PricingRepositoryService
+{
+  protected readonly manager_: SqlEntityManager
+
+  constructor({ manager }: { manager: SqlEntityManager }) {
+    // @ts-ignore
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments)
+
+    this.manager_ = manager
+  }
+
+  async calculatePrices(
+    pricingFilters: PricingFilters,
+    pricingContext: PricingContext = { context: {} },
+    sharedContext: Context = {}
+  ): Promise<CalculatedPriceSetDTO[]> {
+    const manager = this.getActiveManager<SqlEntityManager>()
+    const knex = manager.getKnex()
+    const context = pricingContext.context || {}
+
+    // Quantity is used to scope money amounts based on min_quantity and max_quantity.
+    // We should potentially think of reserved words in pricingContext that can't be used in rules
+    // or have a separate pricing options that accept things like quantity, price_list_id and other
+    // pricing module features
+    const quantity = context.quantity
+    delete context.quantity
+
+    // Currency code here is a required param.
+    const currencyCode = context.currency_code
+    delete context.currency_code
+
+    if (!currencyCode) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        `currency_code is a required input in the pricing context`
```

Comment:
suggestion(non-blocking): Add method to error message
`Method calculatePrices requires currency_code in the pricing context`